rudder-templates-cli/src/main/scala/com/normation/templates/cli/TemplateCli.scala

```diff
@@ -0,0 +1,282 @@
+  , templates     : Seq[File] = Seq()
+  , out           : File      = new File(".")
+  , verbose       : Boolean   = false
+  , inputExtension: String = ".st"
+  , outputExtension: String = ""
+  , showStackTrace: Boolean = false
+)
+
+object Tryor {
+  //the lazy param is of course necessary, else the exception is thrown
+  //before going to the block, never caught.
+  def apply[T](cmd: => T, errorMsg: String): Box[T] = {
+    try {
+      Full(cmd)
+    } catch {
+      case ex: Exception => Failure(s"${errorMsg}: ${ex.getMessage}", Full(ex), Empty)
+    }
+  }
+}
+
+object TemplateCli {
+
+  val fillerService = new FillTemplatesService()
+
+  val parser = new OptionParser[Config]("Rudder template cli") {
+    head("rudder-templates-cli", "3.3.x")
+
+    opt[File]('o', "out") valueName("<file>") action { (x, c) =>
+      c.copy(out = x) } text("output directory for filled template, default is '.'")
+
+    opt[String]("inext") optional() valueName("<input file extension>") action { (x, c) =>
+       c.copy(inputExtension = x) } text("extension of input templates. Default is '.st'")
+
+    opt[String]("outext") optional() valueName("<output file extension>") action { (x, c) =>
+       c.copy(inputExtension = x) } text("extension of templates after processing. Default is '.cf'")
+
+    opt[File]('p', "param") optional() valueName("<variable.json>") action { (x, c) =>
+      c.copy(variables = x) } text("JSON file defining variables. Default is 'variables.json'. See below for format details.")
+
+    opt[Unit]('X', "stackTrace") optional() action { (_, c) =>
+      c.copy(showStackTrace = true) } text("Print stack trace on error")
+
+    arg[File]("<template.st>...") unbounded() action { (x, c) =>
+      c.copy(templates = c.templates:+ x) } text("""list of templates to fill. Only file with the correct extension (by default '.st') will
+        | be processed. The extension will be replaced by '.cf' by default, ounce processed.""".stripMargin)
+
+    help("help") text("prints this usage text")
+
+    note("""The expected format for variables.json is a simple key:value file, with value being only string, boolean or Array of string. 'system' and 'optioannal' properties can also be specified:
+       | {
+       |     "key1": true
+       |   , "key2": "some value"
+       |   , "key3": "42"
+       |   , "key4": [ "some", "more", "values", true, false ]
+       |   , "key5": { "value": "k5", "system": true, "optional": false }
+       |   , "key6": { "value": [ "a1", "a2", "a3" ], "system": false, "optional": true }
+       |   , "key7": ""
+       |   , "key8": { "value": [] }
+       | }
+      """.stripMargin)
+  }
+
+  def main(args: Array[String]): Unit = {
+
+    //in case of error with args, stop and display usage
+    val config = parser.parse(args, Config()).getOrElse {
+        parser.showUsage
+        System.exit(1)
+        //just for type inference, never reached
+        Config()
+    }
+
+
+    val res = for {
```

Comment:
Maybe just need a comment explaining your are filling templates here or just rename res to "filledTemplates"?